applications/callflow/src/cf_endpoint.erl

```diff
@@ -547,6 +581,22 @@ maybe_endpoint_called_self(Endpoint, Properties, Call) ->
         'false' ->
             lager:info("endpoint ~s is calling self", [EndpointId]),
             {'error', 'endpoint_called_self'}
+    end;
```

Comment:
since function above shares exact same bottom half logic as function below - maybe send to helper function?